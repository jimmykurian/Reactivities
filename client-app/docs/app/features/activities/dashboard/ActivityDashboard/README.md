[**Client App Documentation v0.0.0**](../../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../../README.md) / app/features/activities/dashboard/ActivityDashboard

# app/features/activities/dashboard/ActivityDashboard

## Index

### Interfaces

- [Props](interfaces/Props.md)

### Functions

- [default](functions/default.md)
