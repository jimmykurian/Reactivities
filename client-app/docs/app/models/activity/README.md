[**Client App Documentation v0.0.0**](../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../README.md) / app/models/activity

# app/models/activity

## Index

### Interfaces

- [Activity](interfaces/Activity.md)
