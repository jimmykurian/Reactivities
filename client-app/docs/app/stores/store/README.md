[**Client App Documentation v0.0.0**](../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../README.md) / app/stores/store

# app/stores/store

## Index

### Interfaces

- [Store](interfaces/Store.md)

### Variables

- [StoreContext](variables/StoreContext.md)
- [store](variables/store.md)

### Functions

- [useStore](functions/useStore.md)
