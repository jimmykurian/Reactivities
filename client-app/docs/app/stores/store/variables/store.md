[**Client App Documentation v0.0.0**](../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../README.md) / [app/stores/store](../README.md) / store

# Variable: store

> `const` **store**: [`Store`](../interfaces/Store.md)

## Constant

store

## Description

The store instance containing all the application's stores.

## Source

[src/app/stores/store.ts:25](https://github.com/jimmykurian/Reactivities/blob/53b13a08b38b7d13db7685da19b0f30adc1de6b5/client-app/src/app/stores/store.ts#L25)
