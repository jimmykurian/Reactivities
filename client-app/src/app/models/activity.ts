/**
 * @author Jimmy Kurian
 * @name Activity
 */

/**
 * The Activity interface represents the structure of an activity object in the Reactivities application.
 * It includes various properties such as id, title, date, description, category, city, and venue.
 *
 * @interface
 *
 * @property {string} id - The unique identifier for the activity.
 * @property {string} title - The title of the activity.
 * @property {string} date - The date of the activity in ISO string format.
 * @property {string} description - A brief description of the activity.
 * @property {string} category - The category to which the activity belongs.
 * @property {string} city - The city where the activity takes place.
 * @property {string} venue - The specific venue of the activity within the city.
 */
export interface Activity {
  id: string;
  title: string;
  date: string;
  description: string;
  category: string;
  city: string;
  venue: string;
}
