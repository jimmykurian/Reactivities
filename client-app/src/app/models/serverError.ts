/**
 * @file ServerError.ts
 * @author Jimmy Kurian
 * @fileoverview This file defines the `ServerError` interface, which models the structure of server error responses in the Reactivities application.
 */

/**
 * The ServerError interface represents the structure of an error returned from the server.
 * It includes the HTTP status code, an error message, and optional detailed information about the error.
 *
 * @interface ServerError
 *
 * @property {number} statusCode - The HTTP status code associated with the error.
 * @property {string} message - A brief message describing the error.
 * @property {string} details - Additional details or stack trace information about the error (optional).
 */
export interface ServerError {
  statusCode: number;
  message: string;
  details: string;
}
