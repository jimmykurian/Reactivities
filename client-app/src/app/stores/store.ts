/**
 * @author Jimmy Kurian
 * @name Store
 */

import { createContext, useContext } from 'react';
import ActivityStore from './activityStore';

/**
 * Interface representing the structure of the store.
 *
 * @interface Store
 * @property {ActivityStore} activityStore - Instance of ActivityStore for managing activity-related state.
 */
interface Store {
  activityStore: ActivityStore;
}

/**
 * The store instance containing all the application's stores.
 *
 * @constant
 * @type {Store}
 */
export const store: Store = {
  activityStore: new ActivityStore(),
};

/**
 * The React context for the store.
 *
 * @constant
 * @type {React.Context<Store>}
 */
export const StoreContext = createContext(store);

/**
 * Custom hook to use the store in React components.
 *
 * @function
 * @returns {Store} The store instance.
 *
 * @example
 * ```tsx
 * import { useStore } from './store';
 *
 * const MyComponent = () => {
 *   const { activityStore } = useStore();
 *   return <div>{activityStore.title}</div>;
 * };
 * ```
 */
export function useStore(): Store {
  return useContext(StoreContext);
}
