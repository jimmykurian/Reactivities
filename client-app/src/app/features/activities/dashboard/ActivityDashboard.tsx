/**
 * @author Jimmy Kurian
 * @name ActivityDashboard
 */

import { Grid, List } from 'semantic-ui-react';
import { Activity } from '../../../models/activity';
import ActivityList from './ActivityList';

/**
 * Props interface for the ActivityDashboard component.
 *
 * @interface Props
 * @property {Activity[]} activities - Array of activities to be displayed.
 */
export interface Props {
  activities: Activity[];
}

/**
 * ActivityDashboard component.
 *
 * @component
 * @param {Props} props - The properties passed to the component.
 * @returns {JSX.Element} The rendered ActivityDashboard component.
 *
 * @remarks
 * This component serves as the dashboard to display a list of activities.
 * It uses the `Grid.Column` component from Semantic UI to structure the layout
 * and the `List` component to contain the `ActivityList` component.
 *
 * @example
 * ```tsx
 * const activities = [
 *   { id: '1', title: 'Activity 1', date: '2023-12-31', description: 'Description 1', category: 'Category 1', city: 'City 1', venue: 'Venue 1' },
 *   { id: '2', title: 'Activity 2', date: '2024-01-01', description: 'Description 2', category: 'Category 2', city: 'City 2', venue: 'Venue 2' },
 * ];
 *
 * <ActivityDashboard activities={activities} />
 * ```
 */
export default function ActivityDashboard({ activities }: Props) {
  return (
    <Grid.Column width="10">
      <List>
        <ActivityList activities={activities} />
      </List>
    </Grid.Column>
  );
}
