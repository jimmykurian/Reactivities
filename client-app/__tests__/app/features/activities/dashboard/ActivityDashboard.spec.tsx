import { render, screen } from '@testing-library/react';
import { faker } from '@faker-js/faker';
import { useStore } from '../../../../../src/app/stores/store';
import ActivityDashboard from '../../../../../src/app/features/activities/dashboard/ActivityDashboard';
import { makeAutoObservable, runInAction } from 'mobx';
import { Activity } from '../../../../../src/app/models/activity';

// Mock the store context
jest.mock('../../../../../src/app/stores/store', () => ({
  useStore: jest.fn(),
}));

describe('ActivityDashboard', () => {
  // Generate mock activities using Faker.js
  const generateMockActivity = (): Activity => ({
    id: faker.string.uuid(),
    title: faker.lorem.words(3),
    date: faker.date.future().toISOString().split('T')[0],
    description: faker.lorem.sentence(),
    category: faker.lorem.word(),
    city: faker.location.city(),
    venue: faker.location.streetAddress(),
  });

  // Generate mock activities dynamically for other tests
  const mockActivities: Activity[] = Array.from(
    { length: 2 },
    generateMockActivity,
  );

  // Fixed mock activities for snapshot test
  const fixedMockActivities: Activity[] = [
    {
      id: '1',
      title: 'Fixed Title 1',
      date: '2023-12-31',
      description: 'Fixed description 1',
      category: 'Fixed category 1',
      city: 'Fixed city 1',
      venue: 'Fixed venue 1',
    },
    {
      id: '2',
      title: 'Fixed Title 2',
      date: '2024-01-01',
      description: 'Fixed description 2',
      category: 'Fixed category 2',
      city: 'Fixed city 2',
      venue: 'Fixed venue 2',
    },
  ];

  class MockActivityStore {
    activityRegistry: Map<string, Activity>;
    selectedActivity: Activity | undefined = undefined;
    editMode = false;
    loading = false;
    loadingInitial = false;
    deleteActivity = jest.fn();
    selectActivity = jest.fn();
    cancelSelectedActivity = jest.fn();
    openForm = jest.fn();
    closeForm = jest.fn();
    createActivity = jest.fn();
    updateActivity = jest.fn();

    constructor(activities: Activity[]) {
      this.activityRegistry = new Map(
        activities.map((activity) => [activity.id, activity]),
      );
      makeAutoObservable(this);
    }

    get activitiesByDate() {
      return Array.from(this.activityRegistry.values()).sort(
        (a, b) => Date.parse(a.date) - Date.parse(b.date),
      );
    }

    setLoadingInitial = (state: boolean) => {
      this.loadingInitial = state;
    };

    loadActivities = async () => {
      this.setLoadingInitial(true);
      runInAction(() => {
        this.setLoadingInitial(false);
      });
    };
  }

  const createMockStore = (activities: Activity[]) => ({
    activityStore: new MockActivityStore(activities),
  });

  test('renders the ActivityDashboard component', () => {
    // Arrange
    const mockStore = createMockStore(mockActivities);
    (useStore as jest.Mock).mockReturnValue(mockStore);

    // Act
    render(<ActivityDashboard />);

    // Assert
    mockActivities.forEach((activity) => {
      const activityElements = screen.getAllByText(activity.title);
      expect(activityElements.length).toBeGreaterThan(0);
      activityElements.forEach((element) =>
        expect(element).toBeInTheDocument(),
      );
    });
  });

  test('matches snapshot', () => {
    // Arrange
    const mockStore = createMockStore(fixedMockActivities);
    (useStore as jest.Mock).mockReturnValue(mockStore);

    // Act
    const { asFragment } = render(<ActivityDashboard />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
