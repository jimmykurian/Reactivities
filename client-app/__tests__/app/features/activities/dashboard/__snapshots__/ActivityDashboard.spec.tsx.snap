// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityDashboard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui grid"
  >
    <div
      class="ten wide column"
    >
      <div
        class="ui list"
        role="list"
      >
        <div>
          Mocked ActivityList
        </div>
      </div>
    </div>
    <div
      class="six wide column"
    >
      <div
        class="ui large vertical menu"
        style="width: 100%; margin-top: 25px;"
      >
        <div
          class="ui teal attached header"
        >
          <i
            aria-hidden="true"
            class="filter icon"
          />
          <div
            class="content"
          >
            Filters
          </div>
        </div>
        <div
          class="item"
        >
          All Activities
        </div>
        <div
          class="item"
        >
          I'm going
        </div>
        <div
          class="item"
        >
          I'm hosting
        </div>
      </div>
      <div
        class="ui header"
      />
      <div
        class="react-calendar"
      >
        <div
          class="react-calendar__navigation"
        >
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
            type="button"
          >
            «
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
            type="button"
          >
            ‹
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__label"
            style="flex-grow: 1;"
            type="button"
          >
            <span
              class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
            >
              August 2024
            </span>
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
            type="button"
          >
            ›
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
            type="button"
          >
            »
          </button>
        </div>
        <div
          class="react-calendar__viewContainer"
        >
          <div
            class="react-calendar__month-view"
          >
            <div
              style="display: flex; align-items: flex-end;"
            >
              <div
                style="flex-grow: 1; width: 100%;"
              >
                <div
                  class="react-calendar__month-view__weekdays"
                  style="display: flex; flex-wrap: nowrap;"
                >
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  >
                    <abbr
                      aria-label="Monday"
                      title="Monday"
                    >
                      Mon
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  >
                    <abbr
                      aria-label="Tuesday"
                      title="Tuesday"
                    >
                      Tue
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  >
                    <abbr
                      aria-label="Wednesday"
                      title="Wednesday"
                    >
                      Wed
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  >
                    <abbr
                      aria-label="Thursday"
                      title="Thursday"
                    >
                      Thu
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  >
                    <abbr
                      aria-label="Friday"
                      title="Friday"
                    >
                      Fri
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  >
                    <abbr
                      aria-label="Saturday"
                      title="Saturday"
                    >
                      Sat
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  >
                    <abbr
                      aria-label="Sunday"
                      title="Sunday"
                    >
                      Sun
                    </abbr>
                  </div>
                </div>
                <div
                  class="react-calendar__month-view__days"
                  style="display: flex; flex-wrap: wrap;"
                >
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="July 29, 2024"
                    >
                      29
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="July 30, 2024"
                    >
                      30
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="July 31, 2024"
                    >
                      31
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 1, 2024"
                    >
                      1
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 2, 2024"
                    >
                      2
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 3, 2024"
                    >
                      3
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 4, 2024"
                    >
                      4
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 5, 2024"
                    >
                      5
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 6, 2024"
                    >
                      6
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 7, 2024"
                    >
                      7
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 8, 2024"
                    >
                      8
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 9, 2024"
                    >
                      9
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 10, 2024"
                    >
                      10
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 11, 2024"
                    >
                      11
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 12, 2024"
                    >
                      12
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 13, 2024"
                    >
                      13
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 14, 2024"
                    >
                      14
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 15, 2024"
                    >
                      15
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 16, 2024"
                    >
                      16
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 17, 2024"
                    >
                      17
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 18, 2024"
                    >
                      18
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 19, 2024"
                    >
                      19
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 20, 2024"
                    >
                      20
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 21, 2024"
                    >
                      21
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 22, 2024"
                    >
                      22
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 23, 2024"
                    >
                      23
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 24, 2024"
                    >
                      24
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 25, 2024"
                    >
                      25
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 26, 2024"
                    >
                      26
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 27, 2024"
                    >
                      27
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 28, 2024"
                    >
                      28
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 29, 2024"
                    >
                      29
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 30, 2024"
                    >
                      30
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="August 31, 2024"
                    >
                      31
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    type="button"
                  >
                    <abbr
                      aria-label="September 1, 2024"
                    >
                      1
                    </abbr>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
